lib/screens/login_signup_screen.dart

```diff
@@ -0,0 +1,398 @@
+import 'package:chat_app/config/palette.dart';
+import 'package:flutter/material.dart';
+
+class LoginSignupScreen extends StatefulWidget {
+  const LoginSignupScreen({super.key});
+
+  @override
+  State<LoginSignupScreen> createState() => _LoginSignupScreenState();
+}
+
+class _LoginSignupScreenState extends State<LoginSignupScreen> {
+  bool isSignupScreen = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Palette.backgroundColor,
+      body: Stack(
+        children: [
+          Positioned(
+            top: 0.0,
+            left: 0.0,
+            right: 0.0,
+            child: Container(
+              height: 300.0,
+              padding: const EdgeInsets.only(bottom: 50.0),
+              decoration: const BoxDecoration(
+                image: DecorationImage(
+                  image: AssetImage('assets/background.jpeg'),
+                  fit: BoxFit.fill,
+                ),
+              ),
+              child: Center(
+                child: RichText(
+                  text: const TextSpan(
+                    text: 'Welcome to ',
+                    style: TextStyle(
+                      letterSpacing: 1.0,
+                      fontSize: 25.0,
+                      color: Colors.white,
+                    ),
+                    children: [
+                      TextSpan(
+                        text: 'MY CHAT',
+                        style: TextStyle(
+                          letterSpacing: 1.0,
+                          fontSize: 25.0,
+                          color: Colors.white,
+                          fontWeight: FontWeight.bold,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            ),
+          ),
+          AnimatedPositioned(
+            duration: const Duration(milliseconds: 200),
+            curve: Curves.easeIn,
+            top: isSignupScreen ? 390.0 : 330.0,
+            left: 0.0,
+            right: 0.0,
+            child: Center(
+              child: Container(
+                padding: const EdgeInsets.only(top: 45.0),
+                height: 100.0,
+                width: MediaQuery.of(context).size.width - 100,
```

Comment:
```suggestion
width: MediaQuery.sizeOf(context).width - 100,
```

MediaQuery.sizeOf를 활용하면, 화면 사이즈가 바뀔 때만을 감지해 리빌드하는 장점이 있습니다.
https://stackoverflow.com/questions/76474858/with-flutter-3-10-which-is-the-better-way-to-get-the-size-with-mediaquery